mesonbuild/interpreter/interpreterobjects.py

```diff
@@ -610,6 +612,26 @@ def as_shared_method(self, args: T.List[TYPE_var], kwargs: InternalDependencyAsK
             raise InterpreterException('as_shared method is only supported on declare_dependency() objects')
         return self.held_object.get_as_shared(kwargs['recursive'])
 
+    @FeatureNew('dependency.as_json', '1.8.0')
+    @noPosargs
+    @noKwargs
+    def as_json(self, args: T.List[TYPE_var], kwargs: TYPE_kwargs) -> mesonlib.File:
+        d = self.held_object.as_dict(self.interpreter.backend)
+        name = d.pop('name')
+        # TODO: the name of internal dependencies changes
+        output = f'meson_depjson_{name}.json'
+
+        ofile_path = os.path.join(self.interpreter.subdir, output)
```

Comment:
(as seen e.g. in configure_file)

This needs to include self.env.build_dir or else it will write into the current working directory (the directory you ran meson in) instead of the build directory.